File: Python/1049 - Animal.py
l1 = ["vertebrado", "invertebrado"]
l2 = ["ave", "mamifero", "inseto", "anelideo"]
l3 = ["carnivoro", "onivoro", "herbivoro", "hematofago"]
s1,s2,s3 = input(), input(), input()
if s1 == l1[0]:
    if s2 == l2[0]:
        if s3 == l3[0]:
            print("aguia")
        elif s3 == l3[1]:
            print("pomba")
    elif s2 == l2[1]:
        if s3 == l3[1]:
            print("homem")
        elif s3 == l3[2]:
            print("vaca")
elif s1 == l1[1]:
    if s2 == l2[2]:
        if s3 == l3[3]:
            print("pulga")
        elif s3 == l3[2]:
            print("lagarta")
    elif s2 == l2[3]:
        if s3 == l3[3]:
            print("sanguessuga")
        elif s3 == l3[1]:
            print("minhoca")

File: C++/1049 - Animal.cpp
#include <iostream>
#include <vector>

bool get_has_vertebra()
{
    std::string input;
    std::cin >> input;
    if(input == "vertebrado")
    {
        return true;
    }
    return false;
}

std::string get_animal(bool has_vertebra)
{
    std::string data1,data2;
    std::cin >> data1 >> data2;
    if (has_vertebra)
    {
        if(data1 == "ave")
        {
            if(data2 == "carnivoro")
            {
                return "aguia";
            }
            if(data2 == "onivoro")
            {
                return "pomba";
            }
            return "N/A";
        }
        if(data1 == "mamifero")
        {
            if(data2 == "onivoro")
            {
                return "homem";
            }
            if(data2 == "herbivoro")
            {
                return "vaca";
            }
            return "N/A";
        }
    }
    else
    {
        if(data1 == "inseto")
        {
            if(data2 == "hematofago")
            {
                return "pulga";
            }
            if(data2 == "herbivoro")
            {
                return "lagarta";
            }
            return "N/A";
        }
        if(data1 == "anelideo")
        {
            if(data2 == "hematofago")
            {
                return "sanguessuga";
            }
            if(data2 == "onivoro")
            {
                return "minhoca";
            }
            return "N/A";
        }
    }
    return "N/A";
}

int main()
{

    bool has_vertebra = get_has_vertebra();
    std::cout << get_animal(has_vertebra) << std::endl;
    return 0;
}